frontend/src/components/WikiList.js

```diff
@@ -0,0 +1,57 @@
+import React from "react";
+
+class WikiList extends React.Component {
+  constructor(props) {
+    super(props);
+    this.handleWikiChange = this.handleWikiChange.bind(this)
+    this.state = {error: false, wikis: []}
+  }
+
+  handleWikiChange(wikiId) {
+    console.log("Clicked wiki " + wikiId)
+    this.props.handleWikiChange(wikiId)
+  }
+
+  componentDidMount() {
+    fetch("http://localhost:8000/api/w", { crossDomain: true })
+      .then((res) => res.json())
+      .then(
+        (returnedWikis) => {
+            if (Array.isArray(returnedWikis)) {
+              this.setState({wikis: returnedWikis, error: null});
+            } else {
+              this.setState({wikis: [], error: "Invalid response"})
```

Comment:
Lower down, you're using `this.state.error.message` which will be undefined for this case. Maybe make this a real `Error` object? (See my suggestion below, too.)